gradio/blocks.py

```diff
@@ -1331,13 +1333,37 @@ def postprocess_data(
 
         return output
 
+    def handle_streaming_outputs(
+        self, fn_index: int, data: list, session_hash: str | None, run: int | None
```

Comment:
Just wondering, why do you need a separate `run` index? Isn't `session_hash` + `fn_index` enough?